### PR TITLE
RELATED: FET-927 Add styles to display dropdown correctly

### PR DESCRIPTION
**Pull request by plavemar**
- Add mobile device support for AttributeFilterButton component

JIRA: FET-927

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by plavemar** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6146/ : SUCCESS in 22s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/87/ : FAILURE in 15m 50s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6145/ : SUCCESS in 24s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7931/ : SUCCESS in 19m 28s

**Comment by plavemar** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1260/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6147/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1260/ : UNSTABLE in 19m 07s

**Comment by plavemar** [comment]
extended check cypress

File: libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/AttributeFilterBody.tsx [hunk 1]
```diff
@@ -130,9 +137,16 @@ const AttributeFilterBodyCore: React.FC<IAttributeDropdownBodyExtendedProps> = (
     );
 
     const currentWidth = width || 245;
+    const actionsClassnames = cx("gd-dialog-footer dropdown-footer", {
+        "dropdown-footer-mobile": isMobile,
+    });
+    const attributeValuesStyles = {
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/AttributeFilterBody.tsx:143]: We should probably memoize this with useMemo, the isMobile variable is unlikely to change often, so it should help with performance nicely.

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6148/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/88/ : FAILURE in 17m 05s

File: libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/AttributeFilterBody.tsx [hunk 2]
```diff
@@ -130,9 +137,18 @@ const AttributeFilterBodyCore: React.FC<IAttributeDropdownBodyExtendedProps> = (
...
     const currentWidth = width || 245;
+    const actionsClassnames = cx("gd-dialog-footer dropdown-footer", {
+        "dropdown-footer-mobile": isMobile,
+    });
+    const attributeValuesStyles = useMemo(() => {
+        return {
+            width: isMobile ? "auto" : currentWidth,
+            overflow: isMobile ? "hidden" : undefined,
+        };
+    }, [isMobile]);
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/AttributeFilterBody.tsx:148]: The `currentWidth` needs to be in the dependency array as well

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2026

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1770/ : SUCCESS in 1m 56s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2051/ : SUCCESS in 56s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/850/ : SUCCESS in 13m 47s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7935/ : SUCCESS in 17m 13s

**Pull request merged**